instrumentation/opentelemetry_bandit/mix.exs

```diff
@@ -74,7 +73,7 @@ defmodule OpentelemetryBandit.MixProject do
       # dev dependencies
       {:ex_doc, ">= 0.0.0", only: :dev, runtime: false},
       {:excoveralls, "~> 0.18", only: :test},
-      {:bandit, "~> 1.0", only: [:dev, :test], runtime: false},
+      {:bandit, "~> 1.4", only: [:dev, :test], runtime: false},
```

Comment:
The idea being that at least 1.4 is needed for this. Not sure if it would make sense to keep 1.0 dependency and make code a bit smarter, e.g. find `:method` in `meta` or `conn`.

Reply:
I'm going to start on the refactor with the bandit event changes and http semantic conventions which will be a breaking change in itself. 

This PR is really helpful in identifying some of the issues y'all have found with the update. We might need some further tweaks by Mat. Can you aggregate what you've found thus far into #136?